ecp5/CMakeLists.txt

```diff
@@ -26,14 +26,38 @@ else()
     message(STATUS "Trellis install prefix: ${TRELLIS_INSTALL_PREFIX}")
 
     if(NOT TRELLIS_LIBDIR)
+        # The pytrellis library isn't a normal shared library, but rather a native Python library;
+        # it does not follow the normal platform conventions for shared libraries, so we can't just
+        # use find_library() here. Instead, we emulate the useful parts of the find_library() logic
+        # for use with find_path().
+        set(pytrellis_path)
+        foreach(prefix_path ${CMAKE_PREFIX_PATH})
+            list(APPEND pytrellis_path ${prefix_path}/lib)
+            if(CMAKE_LIBRARY_ARCHITECTURE)
+                list(APPEND pytrellis_path ${prefix_path}/lib/${CMAKE_LIBRARY_ARCHITECTURE})
+            endif()
+        endforeach()
+        list(APPEND pytrellis_path ${CMAKE_LIBRARY_PATH})
+        if(NOT NO_CMAKE_SYSTEM_PATH)
+            foreach(prefix_path ${CMAKE_SYSTEM_PREFIX_PATH})
+                list(APPEND pytrellis_path ${prefix_path}/lib)
+                if(CMAKE_LIBRARY_ARCHITECTURE)
+                    list(APPEND pytrellis_path ${prefix_path}/lib/${CMAKE_LIBRARY_ARCHITECTURE})
+                endif()
+            endforeach()
+            list(APPEND pytrellis_path ${CMAKE_SYSTEM_LIBRARY_PATH})
+        endif()
+        message(STATUS "Searching for pytrellis in: ${pytrellis_path}")
+
         if(WIN32)
             set(pytrellis_lib pytrellis.pyd)
         else()
             set(pytrellis_lib pytrellis${CMAKE_SHARED_LIBRARY_SUFFIX})
         endif()
+
         find_path(TRELLIS_LIBDIR ${pytrellis_lib}
             HINTS ${TRELLIS_INSTALL_PREFIX}/lib/${TRELLIS_PROGRAM_PREFIX}trellis
-            PATHS ${CMAKE_SYSTEM_LIBRARY_PATH} ${CMAKE_LIBRARY_PATH}
+            PATHS ${pytrellis_paths}
```

Comment:
Typo - should be `pytrellis_path`

Reply:
The typo was actually the other way around, PTAL.

Reply:
Works for me :+1: